src/v/utils/unresolved_address.h

```diff
@@ -24,13 +24,17 @@
 /// tuple
 class unresolved_address {
 public:
+    using inet_family = std::optional<ss::net::inet_address::family>;
+
     unresolved_address() = default;
-    unresolved_address(ss::sstring host, uint16_t port)
+    unresolved_address(ss::sstring host, uint16_t port, inet_family family = {})
```

Comment:
could simply default to `family::I_NET` - doesn't ip_address automatically detect the family based on the string? 

in my head, there could be a parsing utility that parses out the ip string and then determines the family that can run in the ctor.

Reply:
The host is not necessary an ip, it can be a host name or url, and address family can't be inferred in this case because same host can have both ipv4 and ipv6 addresses. The dns system can have A-record for both addresses and the 'familiy' parameter will help us in this case.

Reply:
Will change the default to ipv4